velox/functions/prestosql/aggregates/tests/ApproxPercentileTest.cpp

```diff
@@ -567,6 +592,64 @@ TEST_F(ApproxPercentileTest, nullPercentile) {
       "Percentile cannot be null");
 }
 
+TEST_F(ApproxPercentileTest, nanPercentile) {
+  vector_size_t floatSize = 10;
+  auto floatValues = makeFlatVector<float>(floatSize, [](auto row) {
+    if (row > 8)
+      return -std::numeric_limits<float>::infinity();
+    else if (row > 7)
+      return std::numeric_limits<float>::infinity();
+    else if (row > 6)
+      return std::nanf("");
+    else
+      return (float)row;
+  });
+
+  testGlobalAgg<float>(
+      floatValues,
+      nullptr,
+      0.05,
+      0.005,
+      -std::numeric_limits<float>::infinity());
+  testGlobalAgg<float>(floatValues, nullptr, 0.11, 0.005, 0.0);
+  testGlobalAgg<float>(floatValues, nullptr, 0.55, 0.005, 4.0);
+  testGlobalAgg<float>(
+      floatValues,
+      nullptr,
+      0.85,
+      0.005,
+      std::numeric_limits<float>::infinity());
+  testGlobalAgg<float>(floatValues, nullptr, 0.95, 0.005, std::nan(""));
+
+  vector_size_t doubleSize = 10;
```

Comment:
the test for double is a copy of the one for floats. Can you create a templatized test so that you can re-use it for either?
for reference see: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/tests/ArrayUnionTest.cpp#L167-L169